swift/apple/FirezoneKit/Sources/FirezoneKit/Managers/VPNProfileManager.swift

```diff
@@ -154,8 +154,19 @@ public class VPNProfileManager {
     // Save the new VPN profile to System Preferences and reload it,
     // which should update our status from nil -> disconnected.
     // If the user denied the operation, the status will be .invalid
-    try await manager.saveToPreferences()
-    try await manager.loadFromPreferences()
+    do {
+      try await manager.saveToPreferences()
+      try await manager.loadFromPreferences()
+    } catch let error as NSError {
+      guard error.domain == "NEVPNErrorDomain",
+            error.code == 5 // permission denied
+      else {
+        throw error
+      }
+
+      // Silence error when the user doesn't click "Allow" on the VPN
+      // permission dialog
```

Comment:
I think this would be easier to read if the condition were flipped: The default cause should be to re-throw the error and the special-case should be to do nothing (perhaps write a debug / info log?)

Reply:
Hm yeah, the issue is that we need to continue in the special case so we'll need an `if-else`.

Reply:
Actually the `self.manager` can be moved inside the `do` so I've fixed this.